los/los.c

```diff
@@ -1,141 +1,107 @@
 #include "los.h"
-#include <stdio.h>
-#include <math.h>
-
-#define TRUE 1
-#define FALSE 0
-
-void step_towards(int *x, int *y, int target_x, int target_y, int diagonal) {
-    int dx = target_x - *x;
-    int dy = target_y - *y;
-
-    if (abs(dx) > abs(dy)) {
-        // need to decrease x
-        *x += (*x < target_x ? 1 : -1);
-    } else if (abs(dx) < abs(dy)) {
-        // need to decrease y
-        *y += (*y < target_y ? 1 : -1);
-    } else if (rand() % 2 == 0) {
-        // pick at random
-        // TODO Sometimes mobs will bump into a wall when they don't need to
-        *x += (*x < target_x ? 1 : -1);
-    } else {
-        *y += (*y < target_y ? 1 : -1);
-    }
+
+bool line_of_sight(level *lvl, int a_x, int a_y, int b_x, int b_y) {
+    //TODO how the hell is is_position_valid working in here?
+    //TODO because I included level.h? yuck!
+    return braise(lvl, a_x, a_y, b_x, b_y, &is_position_valid);
 }
 
-void update(int *stepper, int *bumper, float slope, int step, int bump, float *err) {
-    // calculate where the next point should be
-    // make sure this ends up as a float!
-    float ideal = *bumper + *err + slope;
+bool can_see(level *lvl, mobile *actor, int target_x, int target_y) {
+    // This is between a thing and a position
+    // It just wraps line_of_sight for easier English reading
+    // Making a thing-to-thing function seems too specific
+    return line_of_sight(lvl, actor->x, actor->y, target_x, target_y);
+}
 
+static bool braise(level *lvl, int a_x, int a_y, int b_x, int b_y, checker_func checker) {
```

Comment:
Very funny. Give it a real name.

Reply:
Also this function is three screens long. I shudder to contemplate it's cyclomentric complexity. Slice it into at least three chunks. I wont make you write tests for them but you'd be happier if you did.

Reply:
renamed

Reply:
23 lines shorter!

I don't know whether you're joking about the cyclomatic complexity.

Reply:
Not joking